pkg/ec2provider/ec2provider.go

```diff
@@ -0,0 +1,274 @@
+	// Look up instance from EC2 API
+	output, err := p.ec2.DescribeInstances(&ec2.DescribeInstancesInput{
+		InstanceIds: aws.StringSlice([]string{id}),
+	})
+	if err != nil {
+		p.unsetRequestInQueueForInstanceId(id)
+		return "", fmt.Errorf("failed querying private DNS from EC2 API for node %s: %s ", id, err.Error())
+	}
+	for _, reservation := range output.Reservations {
+		for _, instance := range reservation.Instances {
+			if aws.StringValue(instance.InstanceId) == id {
+				privateDNSName = aws.StringValue(instance.PrivateDnsName)
+				p.setPrivateDNSNameCache(id, privateDNSName)
+				p.unsetRequestInQueueForInstanceId(id)
+			}
+		}
+	}
+
+	if privateDNSName == "" {
+		return "", fmt.Errorf("failed to find node %s ", id)
+	}
+	return privateDNSName, nil
+}
+
+func (p *ec2ProviderImpl) StartEc2DescribeBatchProcessing() {
+	startTime := time.Now()
+	var instanceIdList []string
+	for {
+		var instanceId string
+		select {
+		case instanceId = <-p.instanceIdsChannel:
+			logrus.Infof("Received the Instance Id := %s from buffered Channel for batch processing ", instanceId)
+			instanceIdList = append(instanceIdList, instanceId)
+		default:
+			// Waiting for more elements to get added to the buffered Channel
+			// And to support the for select loop.
+			time.Sleep(20 * time.Millisecond)
+		}
+		endTime := time.Now()
+		/*
+			The if statement checks for empty list and ignores to make any ec2:Describe API call
+			If elements are less than 100 and time of 200 millisecond has elapsed it will make the
+			ec2:DescribeInstances call with as many elements in the list.
+			It is also possible that if the system gets more than 99 elements in the list in less than
+			200 milliseconds time it will the ec2:DescribeInstances call and that's our whole point of
+			optimization here. Also for FYI we have client level rate limiting which is what this
+			ec2:DescribeInstances call will make so this call is also rate limited.
+		*/
+		if (len(instanceIdList) > 0 && (endTime.Sub(startTime).Milliseconds()) > max_wait_interval_for_batch) || len(instanceIdList) > max_instances_batch_size {
+			startTime = time.Now()
+			logrus.Infof(" Making call to getPrivateDnsAndPublishToCache for batch of instances")
+			dupInstanceList := make([]string, len(instanceIdList))
+			copy(dupInstanceList, instanceIdList)
+			go p.getPrivateDnsAndPublishToCache(dupInstanceList)
+			instanceIdList = nil
+		}
+	}
+}
+
+func (p *ec2ProviderImpl) getPrivateDnsAndPublishToCache(instanceIdList []string) {
+	// Look up instance from EC2 API
+	logrus.Infof("Making Batch Query to DescribeInstances for %v instances ", len(instanceIdList))
+	output, err := p.ec2.DescribeInstances(&ec2.DescribeInstancesInput{
+		InstanceIds: aws.StringSlice(instanceIdList),
+	})
+	if err != nil {
+		logrus.Errorf("Batch call failed querying private DNS from EC2 API for nodes [%s] : with error = []%s ", instanceIdList, err.Error())
+	} else {
+		if output.NextToken != nil {
```

Comment:
Can you make the following two logs debug level?

Reply:
Done.